autopilot/hostfilter.go

```diff
@@ -278,19 +277,17 @@ func (c *contractor) isUsableContract(cfg api.AutopilotConfig, ci contractInfo,
 	return
 }
 
-func isOutOfFunds(cfg api.AutopilotConfig, s rhpv2.HostSettings, c api.Contract) bool {
+func isOutOfFunds(cfg api.AutopilotConfig, pt rhpv3.HostPriceTable, c api.Contract) bool {
 	// TotalCost should never be zero but for legacy reasons we check and return
 	// true should it be the case
 	if c.TotalCost.IsZero() {
 		return true
 	}
 
-	blockBytes := types.NewCurrency64(rhpv2.SectorSize * cfg.Contracts.Period)
-	sectorStoragePrice := s.StoragePrice.Mul(blockBytes)
-	sectorUploadBandwidthPrice := s.UploadBandwidthPrice.Mul64(rhpv2.SectorSize)
-	sectorDownloadBandwidthPrice := s.DownloadBandwidthPrice.Mul64(rhpv2.SectorSize)
-	sectorBandwidthPrice := sectorUploadBandwidthPrice.Add(sectorDownloadBandwidthPrice)
-	sectorPrice := sectorStoragePrice.Add(sectorBandwidthPrice)
+	sectorPrice, _ := pt.BaseCost().
+		Add(pt.AppendSectorCost(cfg.Contracts.Period)).
+		Add(pt.ReadSectorCost(rhpv2.SectorSize)).
+		Total()
 	percentRemaining, _ := big.NewRat(0, 1).SetFrac(c.RenterFunds().Big(), c.TotalCost.Big()).Float64()
 
 	return c.RenterFunds().Cmp(sectorPrice.Mul64(3)) < 0 || percentRemaining < minContractFundUploadThreshold
```

Comment:
oh this is probably where the `3` comes from in he collateral calculation too?

Reply:
Haha yeah that's where it's from